### PR TITLE
chashmap: use debug_map

### PR DESCRIPTION
**Pull request by m4b**
uses `debug_map`. This should ~~theoretically~~ enable pretty printing requests from user ~~but I can't seem to get it to work :/~~

~~Maybe someone else can see why?~~

### CODE REVIEW

**Comment by m4b** [comment]
Can now pretty print the hashmap, e.g.:

before, can't pretty print:
```
"hello" => Data { val: 0 }"foo" => Data { val: 1301564913352704000 }"more" => Data { val: 32760 }"bilz" => Data { val: 25107347865600 }"borp" => Data { val: 376610217984000 }"zoom" => Data { val: 149448499200 }"cadabra" => Data { val: 8648640 }"soon" => Data { val: 786240 }"dog" => Data { val: 840 }"food" => Data { val: 15 }"world" => Data { val: 3904694740058112000 }"baz" => Data { val: 54231871389696000 }"zilp" => Data { val: 1046139494400 }"abra" => Data { val: 345945600 }"zam" => Data { val: 9340531200 }"bar" => Data { val: 162695614169088000 }
```
after, regular and pretty:
```
{"baz": Data { val: 3012881743872000 }, "world": Data { val: 216927485558784000 }, "bar": Data { val: 9038645231616000 }, "zam": Data { val: 9340531200 }, "zilp": Data { val: 1046139494400 }, "zoom": Data { val: 149448499200 }, "bilz": Data { val: 25107347865600 }, "borp": Data { val: 376610217984000 }, "foo": Data { val: 72309161852928000 }, "abra": Data { val: 345945600 }, "cadabra": Data { val: 8648640 }, "dog": Data { val: 840 }, "more": Data { val: 32760 }, "soon": Data { val: 786240 }, "hello": Data { val: 0 }, "food": Data { val: 15 }}

{
    "hello": Data {
        val: 0
    },
    "soon": Data {
        val: 786240
    },
    "cadabra": Data {
        val: 8648640
    },
    "borp": Data {
        val: 376610217984000
    },
    "bilz": Data {
        val: 25107347865600
    },
    "dog": Data {
        val: 840
    },
    "food": Data {
        val: 15
    },
    "world": Data {
        val: 216927485558784000
    },
    "zam": Data {
        val: 9340531200
    },
    "more": Data {
        val: 32760
    },
    "baz": Data {
        val: 3012881743872000
    },
    "foo": Data {
        val: 72309161852928000
    },
    "zoom": Data {
        val: 149448499200
    },
    "bar": Data {
        val: 9038645231616000
    },
    "abra": Data {
        val: 345945600
    },
    "zilp": Data {
        val: 1046139494400
    }
}
```

**Comment by m4b** [comment]
Ping!

**Comment by ticki** [comment]
See my review comments. They don't seem to have been completed yet :)

**Comment by m4b** [comment]
Hmmm I don't seem to see any comments? And i definitely didn't get notified if so :/

**Comment by ticki** [comment]
@m4b Very weird. Don't you see these comments:

**Comment by m4b** [comment]
@ticki No. I'm 99% sure that you haven't clicked the "complete review" option (which was added semi-recently to github), which is why the comments say "pending" :)

**Comment by m4b** [comment]
And on that note, what exactly do you mean by fixing the comment?

**Comment by ticki** [comment]
@m4b Oh hah, a few weeks back you could just leave comments without completing the review.

**Comment by m4b** [comment]
Technology: not even once

:D

File: chashmap/src/lib.rs [hunk 1]
```diff
@@ -430,18 +430,19 @@ impl<K: Clone, V: Clone> Clone for Table<K, V> {
 
 impl<K: fmt::Debug, V: fmt::Debug> fmt::Debug for Table<K, V> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        // create a debug map and fill with entries
```
Comment (m4b) [inline chashmap/src/lib.rs]: What precisely do you mean, like capitalize and stuff ?

**Comment by m4b** [comment]
@ticki ping, I sincerely don't know what you'd like fixed exactly with the comments. Let me know and I'll fix it asap, would like to see this merged so can pretty print.

**Comment by ticki** [comment]
I'm so sorry that this have been laying for so long, I've been on vacation and a bunch of other stuff. I'll merge now.

**Comment by m4b** [comment]
No worries at all, hope your vacation was good ! :D

**Pull request merged**